hooks/use-sync-url-params-with-store/index.js

```diff
@@ -0,0 +1,43 @@
+import { useEffect } from 'react'
+import { useRouter } from 'next/router'
+
+export const useSyncUrlParamsWithStore = (params, { url, as = url }) => {
+  const router = useRouter()
+  const setUrlParams = () => {
+    const newParams = Array.from(params.entries()).filter(
+      ([, value]) => value != null
+    )
+
+    const query = Object.fromEntries(newParams)
+
+    router.push(
+      { pathname: url, query },
+      { pathname: as, query },
+      { shallow: true }
```

Comment:
If you use 'shallow' you may just get a URL from the router and don't pass it as a parameter.

Reply:
Yes, that's something I didn't test yet.  Will probably do it as you are suggesting.